pkg/agent/plugin/svidpublisher/aws/aws.go

```diff
@@ -0,0 +1,230 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/gogo/protobuf/proto"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/go-spiffe/v2/proto/spiffe/workload"
+	"github.com/spiffe/spire/pkg/agent/plugin/svidpublisher"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/proto/spire/common"
+	spi "github.com/spiffe/spire/proto/spire/common/plugin"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	pluginName = "aws_secretsmanager"
+)
+
+var (
+	accessKeyID     = os.Getenv("AWS_ACCESS_KEY_ID")
+	secretAccessKey = os.Getenv("AWS_SECRET_ACCESS_KEY")
+	region          = os.Getenv("AWS_REGION")
+)
+
+// TODO: may we provide it from a data []byte? in json format?
+type secret struct {
+	name  string
+	arn   string
+	kmsId string
+}
+
+func BuiltIn() catalog.Plugin {
+	return builtin(New())
+}
+
+func builtin(p *SvidPublisherPlugin) catalog.Plugin {
+	return catalog.MakePlugin(pluginName, svidpublisher.PluginServer(p))
+}
+
+func New() *SvidPublisherPlugin {
+	return &SvidPublisherPlugin{}
+}
+
+type Config struct {
+	AccessKeyID      string `hcl:"access_key_id"`
+	SecretsAccessKey string `hcl:"secret_access_key"`
+	Region           string `hcl:"region"`
+}
+
+type SvidPublisherPlugin struct {
```

Comment:
I think that the name of this struct should be specific for each implementation of the plugin type. In the case of AWS and Google Cloud they name the service as "Secret Manager" so maybe SecretManager should be part of the name. Azure names the service "Key Vault" so in that case the name should contain "KeyVault".